autocertkit/network_tests.py

```diff
@@ -1002,6 +1002,91 @@ def test_ping(self, session):
         return self._run_test(session)
 
 
+class MulticastTestClass(testbase.NetworkTestClass, IperfTest):
+    """A test class for making sure that multicast traffic
+    from a VM to another VM on another host works."""
+
+    username = 'root'
+    password = DEFAULT_PASSWORD
+    MULTICAST_IP = '226.94.1.1'
+    iperf_report_path = '/tmp/iperf.log'
+
+    def _setup_network(self, session):
+        """Utility method for returning the network reference to be used by VMs"""
+
+        management_network_ref = get_management_network(session)
+
+        # Pick a network to use for testing that exercises the device
+        # are wanting to test
+        self.network_for_test = self.get_networks()[0]
+        self.network = self.network_for_test
+        log.debug('Network for testing with: %s'
+                  % self.network_for_test)
+
+        return [management_network_ref, self.network_for_test]
+
+    def _setup_vms(self, session, network_refs):
+        """Util function for returning VMs to run IPerf test on,
+        can be subclassed to run different configurations"""
+
+        log.debug('Setting up VM - VM cross host test')
+
+        # Setup default static manager with the available interfaces
+        sms = {}
+        for network_ref in network_refs:
+            sms[network_ref] = self.get_static_manager(network_ref)
+
+        return deploy_two_droid_vms(session, network_refs, sms)
+
+    def run_iperf_server(self):
+        """Start the iPerf server listening on a VM"""
+
+        log.debug('Starting IPerf server')
+        self.server_ip = self.get_server_ip('eth0')
+        cmd_str = 'nohup iperf -y csv -s -u -B %s </dev/null &>%s &' \
+            % (self.MULTICAST_IP, self.iperf_report_path)
```

Comment:
IperfTest.run_iperf_server() uses '-D' option, that means runs as a daemon, which closes stdout/stderr, so can not save log from output.